.github/workflows/main.yml

```diff
@@ -26,6 +26,14 @@ jobs:
         uses: mamba-org/setup-micromamba@v1
         with:
           environment-file: ${{ matrix.env }}
+          cache-environment: true
+          cache-downloads: true
+          init-shell: bash
+
+      - name: Install pip dependencies and our package
+        shell: bash -leo pipefail {0}
+        run: |
+          python -m pip install ".[all]"
```

Comment:
Nice that's a much more intelligible separation between base environment and package under test.